src/logging.cpp

```diff
@@ -121,7 +142,11 @@ bool BCLog::Logger::DisableCategory(const std::string& str)
 
 bool BCLog::Logger::WillLogCategory(BCLog::LogFlags category) const
 {
-    return (m_categories.load(std::memory_order_relaxed) & category) != 0;
+    if (Assume(category < NONE)) {
+        StdLockGuard scoped_lock(m_cs);
+        return m_categories.test(category);
```

Comment:
For instance, this code can be called frequently with some logging configs IIRC.

Reply:
Thanks, good observation, I tried to continue using atomic, but wasn't able to get it to work.

But also, I think that with current master, it may be possible for two threads to try to enable two different logging categories at the same time, and one silently fails (both threads read, thread 1 writes, thread 2 writes, undoing thread 1's write). Using a mutex as this PR does solves this problem. What do you think? Maybe still better to use `std::atomic`?

Reply:
Maybe this should be a rw lock? You're correct that testing performance is more important, and using rw lock would help with that.

Reply:
As per my footnote on https://github.com/bitcoin/bitcoin/pull/26697#discussion_r1055932559 - if using `std::atomic<uint64_t>` would improve performance, that could be an argument to stop using the bitset altogether?